### PR TITLE
Custom mailer plugin

### PR DESCRIPTION
**Pull request by gmethvin**
@raymondng 

This is mainly for SecureSocial and only implements the subset of things that we support in ElectronicMail.

### CODE REVIEW

File: kifi-backend/modules/common/conf/play.plugins [hunk 1]
```diff
@@ -1,4 +1,4 @@
-1500:com.typesafe.plugin.CommonsMailerPlugin
```
Comment (raymondng) [inline kifi-backend/modules/common/conf/play.plugins:1]: Awesome. Thanks!

File: kifi-backend/modules/common/app/com/keepit/common/mail/EmailAddresses.scala [hunk 2]
```diff
@@ -29,9 +29,10 @@ object EmailAddresses {
   case object RAY extends SystemEmailAddress("ray@42go.com")
   case object CONGRATS extends SystemEmailAddress("congrats@kifi.com")
   case object ASANA_PROD_HEALTH extends SystemEmailAddress("x+7368498674275@mail.asana.com")
+  case object NOTIFY extends SystemEmailAddress("42.notify@gmail.com")
```
Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/common/mail/EmailAddresses.scala:32]: Not sure if we want to include this -- I created this email just for local testing purposes ... probably using some other kifi.com or 42go.com addresses would be more appropriate.

Comment (gmethvin) [inline kifi-backend/modules/common/app/com/keepit/common/mail/EmailAddresses.scala:32]: The issue is the way our `ElectronicMail` model works. It expects to have a `SystemEmailAddress` in the `from` field. Maybe it makes sense to refactor this code to depend on a different class; this was the easiest way to get it working.

File: kifi-backend/modules/shoebox/app/com/keepit/common/mail/PlayMailerPlugin.scala [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+class PlayMailerPlugin(app: Application) extends MailerPlugin {
+  def email: MailerAPI = {
+    val global = app.global.asInstanceOf[FortyTwoGlobal]
+    implicit val injector = global.injector
+    global.inject[PlayMailerAPI]
+  }
+}
+
+private class PlayMailerAPI @Inject()(db: Database, postOffice: LocalPostOffice) extends MailerAPI {
+  private[this] val mail = new DynamicVariable(ElectronicMail(
```
Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/common/mail/PlayMailerPlugin.scala]: Interesting choice. Just for my education -- can you elaborate on the rationale of this approach?

File: kifi-backend/modules/common/app/com/keepit/common/mail/EmailAddresses.scala (hunk 2 above)
Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/common/mail/EmailAddresses.scala:32]: Thanks. Actually we may need a no-reply@42go.com. WDYT?

**Pull request merged**